File: baekjoon/11723.cpp
#define _CRT_SECURE_NO_WARNINGS

#include <bits/stdc++.h>

using namespace std;

int main(void) {
	ios_base::sync_with_stdio(0);
	cin.tie(0);
	cout.tie(0);

	int N;
	cin >> N;

	string str;
	int temp, BIT = 0;
	
	while (N--) {
		cin >> str;
		if (str == "add") {
			cin >> temp;
			BIT |= (1 << temp);
		}
		else if (str == "remove") {
			cin >> temp;
			BIT &= ~(1 << temp);
		}
		else if (str == "check") {
			cin >> temp;
			if (BIT & (1 << temp))
				cout << 1 << '\n';
			else
				cout << 0 << '\n';
		}
		else if (str == "toggle") {
			cin >> temp;
			BIT ^= (1 << temp);
		}
		else if (str == "all") {
			BIT = (1 << 21) - 1;
		}
		else if (str == "empty") {
			BIT = 0;
		}
	}


	return 0;
}

File: baekjoon/11723.py
import sys
S = set()
total = set()
for i in range(1, 21):
    total.add(i)
M = int(sys.stdin.readline())
for _ in range(M):
    temp = sys.stdin.readline().strip().split()

    if temp[0] == "add":
        S.add(int(temp[1]))

    elif temp[0] == "remove":
        # remove 사용시 없는 것을 지울려고 하면 keyError
        S.discard(int(temp[1]))

    elif temp[0] == "check":
        if int(temp[1]) in S:
            print(1)
        else:
            print(0)

    elif temp[0] == "toggle":
        if int(temp[1]) in S:
            S.discard(int(temp[1]))
        else:
            S.add(int(temp[1]))

    elif temp[0] == "all":
        S = total

    elif temp[0] == "empty":
        S = set()
